modules/invoicer_invoice/src/Form/InvoicerInvoiceConfigForm.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\invoicer_invoice\Form\InvoicerInvoiceConfigForm.
+ */
+
+namespace Drupal\invoicer_invoice\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Configure file system settings for this site.
+ */
+class InvoicerInvoiceConfigForm extends ConfigFormBase {
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFormId() {
+        return 'invoicer_invoice_config_form';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getEditableConfigNames() {
+        return ['invoicer_invoice.settings'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(array $form, FormStateInterface $form_state) {
+        $config = $this->config('invoicer_invoice.settings');
+
+        $form = array();
+
+        $form['vat_types'] = array(
+            '#type' => 'textarea',
+            '#title' => t('VAT types'),
+            '#size' => 60,
+            '#maxlength' => 60,
+            '#default_value' => implode("\n", $config->get('vat_types')),
+            '#description' => t('The VAT types'),
+        );
+
+        return parent::buildForm($form, $form_state);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function submitForm(array &$form, FormStateInterface $form_state) {
+
+        $config = $this->config('invoicer_invoice.settings');
+
+        $form_state->cleanValues();
+        foreach ($form_state->getValues() as $key => $value) {
```

Comment:
Look at:
https://api.drupal.org/api/drupal/core!modules!options!src!Plugin!Field!FieldType!ListItemBase.php/class/ListItemBase/8.2.x

```
 protected static function extractAllowedValues($string, $has_data) {
    $values = array();

    $list = explode("\n", $string);
    $list = array_map('trim', $list);
    $list = array_filter($list, 'strlen');

    $generated_keys = $explicit_keys = FALSE;
    foreach ($list as $position => $text) {
      // Check for an explicit key.
      $matches = array();
      if (preg_match('/(.*)\|(.*)/', $text, $matches)) {
        // Trim key and value to avoid unwanted spaces issues.
        $key = trim($matches[1]);
        $value = trim($matches[2]);
        $explicit_keys = TRUE;
      }
      // Otherwise see if we can use the value as the key.
      elseif (!static::validateAllowedValue($text)) {
        $key = $value = $text;
        $explicit_keys = TRUE;
      }
      // Otherwise see if we can generate a key from the position.
      elseif (!$has_data) {
        $key = (string) $position;
        $value = $text;
        $generated_keys = TRUE;
      }
      else {
        return;
      }

      $values[$key] = $value;
    }

    // We generate keys only if the list contains no explicit key at all.
    if ($explicit_keys && $generated_keys) {
      return;
    }

    return $values;
  }
```